### PR TITLE
Add docstrings for compress/decompress. 

### LINKED ISSUE #26: No documentation for compress or decompress
Seems trivial, but it would be helpful to have some basic documentation for `pyzfp.compress` and `pyzfp.decompress`. From IPython on Jupyter, it should be possible to type e.g. `?pyzfp.compress` and get some basic information about parameters required and how they work. Right now I get:

```
In [3]: ?pyzfp.compress
Docstring: <no docstring>
Type:      builtin_function_or_method

In [4]: ?pyzfp.decompress
Docstring: <no docstring>
Type:      builtin_function_or_method
```

### PR DESCRIPTION
**Pull request by navjotk**
Also raise error for parallel decompress.

Fixes #26

### CODE REVIEW

**Comment by navjotk** [comment]
@nkemnitz Do you mind doing a quick sanity check of the documentation that this PR is adding? Thanks!

**Comment by navjotk** [comment]
@william-silversmith mind being a second set of eyes to review this documentation I'm adding?

File: pyzfp.pyx [hunk 1]
```diff
@@ -170,6 +170,34 @@ cdef zfp_field* init_field(np.ndarray indata):
...
+        One of tolerance, precision, or rate must be specified.
+    precision : int, optional
+        The precision of the compressed data.
+        This will use ZFP in fixed-precision mode.
+        One of tolerance, precision, or rate must be specified.
+    rate : float, optional
+        The rate of the compressed data.
+        This will use ZFP in fixed-rate mode.
+        One of tolerance, precision, or rate must be specified.
+    parallel : bool, optional
```
Comment (william-silversmith) [inline pyzfp.pyx]: Should note the default is `True`. I've found that Cython doesn't let you see the function sig so the documentation has to be manually added (or you can add a pure python wrapper around the cython call).

File: pyzfp.pyx [hunk 2]
```diff
@@ -170,6 +170,34 @@ cdef zfp_field* init_field(np.ndarray indata):
...
+    """
+    Compress a numpy array using zfp.
+
+    Parameters
+    ----------
+    indata : numpy.ndarray
+        The data to compress.
+    tolerance : float, optional
+        The tolerance for the compressed data.
+        This will use ZFP in fixed-tolerance mode.
```
Comment (william-silversmith) [inline pyzfp.pyx]: I think the zfp documentation calls this "fixed-accuracy" mode. Maybe mention that? Maybe also link to the documentation here.

https://zfp.readthedocs.io/en/latest/modes.html#mode-fixed-accuracy

File: pyzfp.pyx [hunk 3]
```diff
@@ -204,7 +232,43 @@ def compress(indata, tolerance=None, precision=None, rate=None, parallel=True):
 
 
 def decompress(const unsigned char[::1] compressed, shape, dtype, tolerance=None, precision=None,
-               rate=None, parallel=True, order='C'):
+               rate=None, parallel=False, order='C'):
```
Comment (william-silversmith) [inline pyzfp.pyx:238]: If you like PR #37 much of this docstring will become obsolete.

File: pyzfp.pyx [hunk 4]
```diff
@@ -204,7 +232,43 @@ def compress(indata, tolerance=None, precision=None, rate=None, parallel=True):
...
+    ----------
+    compressed : Cython.memoryview
+        The compressed data.
+    shape : tuple
+        The shape of the decompressed data.
+    dtype : numpy.dtype
+        The dtype of the decompressed data.
+    tolerance : float, optional
+        The tolerance for the compressed data.
+        This will use ZFP in fixed-tolerance mode.
```
Comment (william-silversmith) [inline pyzfp.pyx]: mention fixed-accuracy mode

File: pyzfp.pyx [hunk 5]
```diff
@@ -204,7 +232,43 @@ def compress(indata, tolerance=None, precision=None, rate=None, parallel=True):
...
+        One of tolerance, precision, or rate must be specified.
+    precision : int, optional
+        The precision of the compressed data.
+        This will use ZFP in fixed-precision mode.
+        One of tolerance, precision, or rate must be specified.
+    rate : float, optional
+        The rate of the compressed data.
+        This will use ZFP in fixed-rate mode.
+        One of tolerance, precision, or rate must be specified.
+    parallel : bool, optional
```
Comment (william-silversmith) [inline pyzfp.pyx]: mention default value

File: pyzfp.pyx (hunk 1 above)
Comment (navjotk) [reply pyzfp.pyx]: Did a quick test with `help(pyzfp.compress)` and it seemed to work fine.

**Pull request merged**